devito/operator.py

```diff
@@ -194,21 +198,16 @@ def signature(self):
                                     if param not in self.input_params]
 
     def apply(self, debug=False):
-        """:param debug: If True, use Python to apply the operator. Default False.
-
-        :returns: A tuple containing the values of the operator outputs
+        """
+        :param debug: If True, use Python to apply the operator. Default False.
+        :returns: A tuple containing the values of the operator outputs or compiled function and its args
         """
         if debug:
             return self.apply_python()
 
-        for param in self.input_params:
-            if hasattr(param, 'initialize'):
-                param.initialize()
+        self.propagator.run(self.get_args())
 
-        args = [param.data for param in self.signature]
-        self.propagator.run(args)
-
-        return tuple([param for param in self.output_params])
+        return tuple([param.data for param in self.output_params])
```

Comment:
This line is the one breaking your branch.
Replace by ` return tuple([param for param in self.output_params])` as it is in master